src/3d/spread3d_wrapper.cu

```diff
@@ -644,6 +684,7 @@ int cuspread3d_blockgather_prop(int nf1, int nf2, int nf3, int M,
 	checkCudaErrors(cudaMemcpy(&totalNUpts,&d_binstartpts[n],
 		sizeof(int),cudaMemcpyDeviceToHost));
 	checkCudaErrors(cudaMalloc(&d_idxnupts,totalNUpts*sizeof(int)));
+	checkCudaErrors(cudaMemset(d_idxnupts,-1,totalNUpts*sizeof(int)));
```

Comment:
Why this change?

Reply:
That is actually for debugging so that I can make sure all entries that should be set are set. I make it a debug statement in commit 2c6909b.

Reply:
Ok.